app/scripts/grading/grade_submission.py

```
@@ -0,0 +1,1734 @@
+  - Student covers all items adequately (no deep detail needed) → checklist_pct ~92–95%
+  - Student covers 80% of items well, rest partially → checklist_pct ~88–90%
+  - Student covers most items but misses 2 entirely → checklist_pct ~78–82%
+  - Student covers about half the items → checklist_pct ~65–70%
+  - Only drop below 60% if the student clearly missed most of the items
+  - Do NOT penalize for phrasing, style, or diagram format choice
+
+Step 3 — Compute checklist_pct:
+  checklist_pct = (yes_count + 0.67 × partial_count) / total_items × 100
+
+Step 4 — Apply GRADE BAND TABLE to get awarded_points:
+  95–100% → multiply max_points by 1.000
```

Ruff (0.15.5)

[warning] String contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF001)